artcommon/artcommonlib/util.py

```diff
@@ -170,3 +172,25 @@ def get_ocp_version_from_group(group):
     if not match:
         raise ValueError(f"Invalid group name: {group}")
     return int(match[1]), int(match[2])
+
+
+def deep_merge(dict1, dict2):
+    """
+    Recursively merge two dictionaries.
+
+    Returns:
+    A new dictionary with merged values.
+    """
+
+    merged = Model(dict1.copy())
+    model2 = Model(dict2)
+
+    for key, value in dict2.items():
+        if isinstance(merged.get(key), dict) and isinstance(value, dict):
+            # If the key is in preserve_keys, and both values are dictionaries, merge them recursively
+            merged[key] = deep_merge(merged[key], value)
+        else:
+            # Otherwise, simply update the value
+            merged[key] = value
+
+    return Model(merged)
```

Comment:
it being a Model doesn't seem important to the function, and the callers re-create it as a Model anyway (which does a deep copy), so why bother here?
```suggestion
    merged = dict1.copy()

    for key, value in dict2.items():
        if isinstance(merged.get(key), dict) and isinstance(value, dict):
            # If both values are dictionaries, merge them recursively
            merged[key] = deep_merge(merged[key], value)
        else:
            # Otherwise, simply update the value
            merged[key] = value

    return merged
```